### PR TITLE
Mark almost all strings for translation

### PR DESCRIPTION
**Pull request by bk2204**
This PR marks almost all of our strings for localization.  In addition, it provides documentation to help contributors of all sorts more gracefully deal with translations and translated strings.

Finally, there is a fake locale which can be created for testing, `i-reverse`, to allow testing whether strings are localized or not.  It consists of the English text, but each word (as in `\w+`) reversed.  This is not built in by default because it would bloat the binary for little benefit.

Most of this series is merely boring invocations of adding `tr.Tr.Get` (or sometimes, `tr.Tr.GetN`).  Occasionally, there are more interesting conversions, such as the need to move variables defined at file scope into functions to be translated, since the locale object will not have been properly initialized until after startup (and the file-scope strings would be run through the locale object before `main` is run).

In most places, we turn a `fmt.Errorf(...)` into an `errors.New(tr.Tr.Get(...))` so that we don't evaluate the translated string a second time.

There is one particular matter which is of interest.  The `Get` function is not a complete analogue of `fmt.Sprintf`.  Unlike `fmt.Sprintf`, `Get` does not process formatting patterns unless there are arguments.  That is, `Get("%s")` returns `%s`, and `Get("%s", "foo")` returns `foo`.  _However_, the linter built into Go will complain if we do `Get("%s")`, since it thinks an argument will be needed, and this causes `make test` to fail.

Our solution is that in a couple places in the code when we need a format string to be localized, we write something like `f := Get("%%s")`, and then later on do `fmt.Sprintf(f, "arg")`.  This looks bizarre, and it is, and I've filed leonelquinteros/gotext#57 about it, so hopefully we can upgrade and then fix this in the future.  However, for now, we'll just go with it.

Due to the amount of commits in this series, boring as most of them might be, I fully expect that this won't receive a review until 2022, and that's fine.  There's no rush.

### CODE REVIEW

**Review by chrisd8088 — CHANGES_REQUESTED** [review]
This is a huge effort, thank you so much!

I think my suggestions and concerns fall into a couple of areas:
- I believe there may be some strings in the `commands/*` source files outside of the `commands/command_*.go` files, such as `commands/uploader.go` and `commands/lockverifier.go`.  These should probably also be converted to the new translation format.
- I'm uncertain if it's correct that the `xgotext` output file, `po/default.pot`, uses backticks to delimit multi-line strings.  If by chance that's a problem in `xgotext`, we should probably submit a patch for that upstream, and then we wouldn't need as much parsing logic in the Ruby helper script.
- Because this PR highlights all our strings, I noticed that there's a some heterogeneity to the prefixes we use, particular.  Some error strings start with `fatal:` but most do not; while warnings may start with `warning:`, `Warning:`, or `WARNING:` (and there's also `ERROR:` in some places).  We should probably take this opportunity to make these all consistent so we aren't giving potential translators extra challenges.
- The `commands/command_*.go` files have many help-text strings for command-line options, which are passed to the Cobra command-line flag functions like `cmd.Flags().StringVarP()`.  But I have a suspicion that none of these are actually available to users, and if that proves to be true, I think maybe we could treat them more like internal comments and save translators the trouble of tackling all of them.

I think those were the main things I noticed, but there are at least two where I'm a bit uncertain and would appreciate a second opinion!

Thank you again so much for kicking off these big projects and getting them to a workable state!

File: docs/l10n.md
Comment (bk2204) [reply docs/l10n.md:51]: I cannot definitively say if there are languages with singular zero as well as multiple plural forms, although I'm not personally aware of any, but I will rephrase as you suggested to avoid being accidentally wrong.

File: commands/command_ls_files.go
Comment (bk2204) [reply commands/command_ls_files.go:31]: Yes, I think this makes sense.  Git has a function called `die`, which prints “fatal:” before its message, so we should also do the same thing and avoid printing it in the message itself.  We can add support for adding “fatal:” in a future series.

File: commands/command_checkout.go
Comment (bk2204) [reply commands/command_checkout.go:44]: I think a followup PR would be very useful for this kind of thing.

File: commands/command_checkout.go
Comment (bk2204) [reply commands/command_checkout.go:173]: I believe you're correct here, and I've removed these in a giant series of fixup commits.

File: commands/command_track.go
Comment (bk2204) [reply commands/command_track.go:235]: I'll remove the second string from translation.  I think it's fine to leave the formatting strings in the former.

File: script/gen-i-reverse
Comment (bk2204) [reply script/gen-i-reverse:50]: Yes, I've filed leonelquinteros/gotext#59 to get this fixed.  It's currently required to make the code work without a lot of complexity, but we can definitely change it when `xgotext` is fixed.

File: config/git_fetcher.go
Comment (bk2204) [reply config/git_fetcher.go:109]: If we're going to do “fatal:” in the future, then let's do “warning:”, since that's the same casing.

**Comment by bk2204** [comment]
Okay, except where specified above, I've made all the changes you suggested in a series of fixup commits, plus a few new commits for the files I've missed.  When you're happy with things, I can squash them down.

File: tq/custom.go
Comment (chrisd8088) [reply tq/custom.go:305]: I think maybe there's still a lowercase `oid` here?

File: script/gen-i-reverse
Comment (chrisd8088) [reply script/gen-i-reverse:50]: Sounds good, thank you for the clarification and the bug report!

File: commands/command_push.go
Comment (chrisd8088) [inline commands/command_push.go:58]: Out of interest, is it possible to trigger this message?  From a cursory glance it seems like, if `len(args) < 1`, then it must be zero, which would be caught above where `Specify a remote and a remote branch name` is output.  Maybe this block could be removed?

**Review by chrisd8088 — CHANGES_REQUESTED** [review]
I tried to go through and mark as many conversations resolved as possible, and I've added a few additional notes, but not many!

The main thing I noticed on a second read-through is that we're a bit wobbly in how we capitalize, or don't, the first word of many messages.  I'm not sure what to say about that and if it's worth tackling here or not.  It obviously doesn't matter to the functionality of the client; it's more a question, perhaps, of whether we want to provide a reliable guide to translators as to whether they should follow suit or not.

**Review by chrisd8088 — APPROVED** [review]
This is definitely looking good!  I went through and added a last bunch of notes (some of which you might want to just ignore), but I didn't want to hold this up asking for another re-review round.

File: lfshttp/client.go
Comment (bk2204) [reply lfshttp/client.go:116]: I think that's implicit enough.

File: tq/transfer_queue.go
Comment (bk2204) [reply tq/transfer_queue.go:952]: I think in this case, since it requires translation, we can just leave it.

**Comment by bk2204** [comment]
I've made all those changes and squashed them down and I'll merge once this is green.

**Pull request merged**
